build.sbt

```diff
@@ -9,12 +9,42 @@ lazy val scalaTestVersion         = "3.0.0"
 lazy val scalaCheckVersion        = "1.13.1"
 lazy val http4sVersion            = "0.15.2a"
 
+enablePlugins(GitVersioning)
+
+git.uncommittedSignifier in ThisBuild := Some("UNCOMMITTED")
+
+// Before printing the prompt check git to make sure all is well.
+shellPrompt in ThisBuild := { state =>
+  import scala.sys.process._
+  import scala.Console.{ RED, RESET }
+  try {
+    val revision = "git rev-parse HEAD".!!.trim
+    val dirty    = "git status -s".!!.trim.length > 0
+    val expected = revision + git.uncommittedSignifier.value.filter(_ => dirty).fold("")("-" + _)
+    val actual   = version.value
+    val stale    = expected != actual
+    if (stale) {
+      print(RED)
+      println(s"Computed version doesn't match the filesystem anymore.")
+      println(s"Please `reload` to get back in sync.")
+      print(RESET)
+    }
+  } catch {
+    case e: Exception =>
+      print(RED)
+      println(s"Couldn't run `git` to check on versioning. Something is amiss.")
+      print(RESET)
+  }
+  "> "
+}
+
 lazy val testLibs = Seq(
   "org.scalatest"  %% "scalatest"  % scalaTestVersion  % "test",
   "org.scalacheck" %% "scalacheck" % scalaCheckVersion % "test"
 )
 
 lazy val commonSettings = Seq(
+  scalaOrganization := "org.typelevel",
```

Comment:
No downside. It gives us `-Ypartial-unification` in 2.11.8 which will be in the mainline compiler in 2.11.9 and we can switch back if we want. This flag enables the fix for SI-2717 which ends up fixing a lot of inference problems.